docs/user-guide/horizontal-pod-autoscaling/index.md

```diff
@@ -22,19 +23,41 @@ to match the observed average CPU utilization to the target specified by user.
 
 ![Horizontal Pod Autoscaler diagram](/images/docs/horizontal-pod-autoscaler.svg)
 
-The autoscaler is implemented as a control loop.
-It periodically queries CPU utilization for the pods it targets.
+The autoscaler is implemented as a control loop
 (The period of the autoscaler is controlled by `--horizontal-pod-autoscaler-sync-period` flag of controller manager.
 The default value is 30 seconds).
-Then, it compares the arithmetic mean of the pod's CPU utilization with the target and adjust the number of replicas if needed.
 
-CPU utilization is the recent CPU usage of a pod divided by the sum of CPU requested by the pod's containers.
-Please note that if some of the pod's containers do not have CPU request set,
-CPU utilization for the pod will not be defined and the autoscaler will not take any action.
-Further details of the autoscaling algorithm are given [here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/horizontal-pod-autoscaler.md#autoscaling-algorithm).
+For each HorizontalPodAutoscaler, the controller fetches the metrics that
+it targets from either the resource metrics API (for per-pod resource metrics),
+or the custom metrics API (for all other metrics).
 
-The autoscaler uses heapster to collect CPU utilization.
-Therefore, it is required to deploy heapster monitoring in your cluster for autoscaling to work.
+For per-pod resource metrics (like CPU), the controller fetches the metrics
+from the resource metrics API for each pod targeted by the HorizontalPodAutoscaler.
+Then, if a target utilization value is set, the controller calculates the utilization
+value as a percentage of the equivalent resource request on the containers in
+each pod.  If a target raw value is set, the raw metric values are used directly.
+It then takes the mean of the utilization or the raw value (depending on the type
+of target specified) across all targeted pods, and produces a ratio used to scale
+the number of desired replicas.
+
+Please note that if some of the pod's containers do not have the relevant resource request set,
+CPU utilization for the pod will not be defined and the autoscaler will not take any action
+for that metric. Further details of the autoscaling algorithm are given [here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/horizontal-pod-autoscaler.md#autoscaling-algorithm).
```

Comment:
"Further details..." -> "See the [autoscaling algorithm design document] for further details about how the autoscaling algorithm works."